include/drivers/rf.h

```diff
@@ -0,0 +1,302 @@
+	/* Binary-phase-shift keying */
+	RF_MODULATION_2PSK,
+	/* Minimum-shift keying/Offset quadrature phase-shift keying */
+	RF_MODULATION_MSK,
+	/* Gaussian minimum-shift keying */
+	RF_MODULATION_GMSK,
+	/* Direct-sequence spread spectrum */
+	RF_MODULATION_DSSS,
+};
+
+/**
+ * @typedef rf_baud_t
+ * @brief Provides a type for setting/getting RF_DEVICE_BAUDRATE
+ */
+typedef uint32_t rf_baud_t;
+
+/**
+ * @typedef rf_power_t
+ * @brief Provides a type for setting/getting RF_DEVICE_OUTPUT_POWER
+ */
+typedef int16_t rf_power_t;
+
+/**
+ * @brief RF device operating modes
+ */
+enum rf_op_mode {
+	/* Perform (self-)calibration of the device */
+	RF_MODE_CALIBRATE = 0,
+	/* Power off the device */
+	RF_MODE_POWER_OFF,
+	/* Put the device to sleep */
+	RF_MODE_SLEEP,
+	/* Idle state of the rf device with shortest possible transistion time to RX/TX */
+	RF_MODE_IDLE,
+	/* Put device to low power state and activate RX on an event */
+	RF_MODE_RX_WAKE_ON_EVENT,
+	/* Put device to low power and periodically wake up to listen for incomnig data */
+	RF_MODE_RX_WAKE_PERIODIC,
+	/* Put the device into RX */
+	RF_MODE_RX,
+	/* Put the device into TX */
+	RF_MODE_TX,
+};
+
+/**
+ * @brief RF events
+ */
+enum rf_event {
+	RF_EVENT_STUCK = 0,
+	RF_EVENT_RECV_READY,
+	RF_EVENT_SEND_READY,
+	RF_EVENT_RECV_DONE,
+	RF_EVENT_SEND_DONE,
+	RF_EVENT_CHANNEL_CLEAR,
+	RF_EVENT_WAKEUP,
+};
+
+/**
+ * @brief RF hardware filter
+ */
+enum rf_hw_filter {
+	RF_HW_FILTER_NONE = 0,
+	RF_HW_FILTER_CRC8,
+	RF_HW_FILTER_CRC16,
+	RF_HW_FILTER_ADDR,
+	RF_HW_FILTER_LENGTH,
+};
+
+/**
+ * @brief RF device set/get arguments
+ */
+enum rf_device_arg {
+	RF_DEVICE_FREQUENCY = 0,
+	RF_DEVICE_CHANNEL,
+	RF_DEVICE_MODULATION_FORMAT,
+	RF_DEVICE_BAUDRATE,
+	RF_DEVICE_OUTPUT_POWER,
+	RF_DEVICE_OPERATING_MODE,
+	RF_DEVICE_SET_EVENT_CB,
+	RF_DEVICE_HW_FILTER,
+	RF_DEVICE_SETTINGS,
+	RF_DEVICE_CALIBRATION_SETTINGS,
+	RF_DEVICE_POWER_TABLE,
+};
+
+/**
+ * @typedef rf_event_cb_t
+ * @brief Callback handler for RF events
+ */
+typedef void (*rf_event_cb_t)(const struct device *dev,
+				enum rf_event event,
+				void *event_params);
+
+/**
+ * @typedef rf_device_set_t
+ * @brief API for setting device parameters
+ */
+typedef int (*rf_device_set_t)(const struct device *dev,
+				    enum rf_device_arg cfg,
+				    void *val);
+
+/**
+ * @typedef rf_device_get_t
+ * @brief API for getting device parameters
+ */
+typedef int (*rf_device_get_t)(const struct device *dev,
+				    enum rf_device_arg cfg,
+				    void *val);
+
+/**
+ * @typedef rf_device_send_t
+ * @brief API for synchronously sending data
+ */
+typedef int (*rf_device_send_t)(const struct device *dev,
+				     union rf_packet *pkt);
+
+/**
+ * @typedef rf_device_recv_t
+ * @brief API for synchronously receiving data
+ */
+typedef int (*rf_device_recv_t)(const struct device *dev,
+				     union rf_packet *pkt);
+
+/**
+ * @brief Callback handler for RF wake event
+ */
+__subsystem struct rf_driver_api {
+	rf_device_set_t device_set;
```

Comment:
on all API function: 
s/rf_device_/rf_

This is just pretty obvious that if you access a struct dev through such API, I it will mean an RF device.